### PR TITLE
Imported Textmesh Pro Essentials

### PR DESCRIPTION
**Pull request by Acimaz**
To reduce the code line changes in https://github.com/UnityTechnologies/open-project-1/pull/30 I created this separate PR which imports the TMPro Essentials.

Note that we are not using the latest version (2.1.1) of TMPro atm We are using the 2.0.1 package, which was in the project already before I imported the essentials. Want me to do an update to the latest verified version?

### CODE REVIEW

**Comment by MileyHollenberg** [comment]
Upgrading to the latest version would be nice yes :), thanks for separating the PRs!

**Comment by Acimaz** [comment]
Just made the upgrade. Settings.json and package-lock.json were created automatically after the upgrade.

File: UOP1_Project/ProjectSettings/Packages/com.unity.probuilder/Settings.json [hunk 1]
```diff
@@ -32,6 +32,36 @@
...
+            },
+            {
+                "type": "UnityEngine.ProBuilder.RectSelectMode, Unity.ProBuilder, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null",
+                "key": "editor.dragSelectRectMode",
+                "value": "{\"m_Value\":0}"
+            },
+            {
+                "type": "UnityEngine.ProBuilder.SelectMode, Unity.ProBuilder, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null",
+                "key": "editor.selectMode",
+                "value": "{\"m_Value\":1}"
```
Comment (MileyHollenberg) [inline UOP1_Project/ProjectSettings/Packages/com.unity.probuilder/Settings.json:64]: I don't think this should have been committed here, doesn't seem to be related to TMPro

**Comment by MileyHollenberg** [comment]
Just had a chat with @ciro-unity and even though I requested you to make these changes I shouldn't have done that as we want to keep package updates and additions to the Unity team itself. I apologize for the miscommunication here on my end

**Comment by Acimaz** [comment]
No problem. It's something which has to be done anyway. Especially when implementing stuff like Settings Systems which depend on UI. I can remove those lines from the JSON, but they were created while upgrading TMPro. So I didn't touch it as I thought it might be related in some way and doesn't harm.

**Comment by ciro-unity** [comment]
Hey @Acimaz and @MileyHollenberg. Miley is correct, we have it in the guidelines that we don't accept PRs to *just* bump project or package versions.
However, I feel like TMPro is a bit of a unique one since it requires new assets in the project (the "essentials") and you couldn't create the functionality without dragging them in.

Also weird that upgrading TMPro changed the settings of ProBuilder. Will note it down.

Let us check the functionality locally, and we'll approve everything.

Thanks both.

**Comment by Acimaz** [comment]
Just removed those few lines in the Settings.json. And yes that's weird :D

**Review by MileyHollenberg — APPROVED** [review]
Tested locally, didn't break anything or throw unwanted warnings. LGTM

**Comment by Acimaz** [comment]
Anything here which needs to be done, beside the merge?

**Comment by Acimaz** [comment]
Looks like ciro has made some changes to the packages in a separate commit which conflict with this PR. Maybe import the essentials straight away then we can close this PR completely.
